temporian/implementation/numpy/data/event.py

```diff
@@ -25,81 +25,148 @@
     int: DType.INT64,
 }
 
+DTYPE_MAPPING = {
+    np.float64: DType.FLOAT64,
+    np.float32: DType.FLOAT32,
+    np.int64: DType.INT64,
+    np.int32: DType.INT32,
+    np.str_: DType.STRING,
+    np.string_: DType.STRING,
+    np.bool_: DType.BOOLEAN,
+}
+DTYPE_REVERSE_MAPPING = {
+    DType.FLOAT64: np.float64,
+    DType.FLOAT32: np.float32,
+    DType.INT64: np.int64,
+    DType.INT32: np.int32,
+    DType.STRING: np.str_,
+    DType.BOOLEAN: np.bool_,
+}
+
+
+@dataclass
+class IndexData:
+    features: List[np.ndarray]
+    timestamps: np.ndarray
+
+    def __init__(
+        self,
+        features: Union[np.ndarray, List[np.ndarray]],
+        timestamps: np.ndarray,
+    ) -> None:
+        if isinstance(features, np.ndarray):
+            features = [features]
+
+        self.features = features
+        self.timestamps = timestamps
+
+    def __len__(self) -> int:
+        return len(self.timestamps)
+
 
 class NumpyEvent:
+    # tolerance levels for comparing float features in __eq__
+    rtol: float = 1e-9
```

Comment:
What about having a "Near" function instead, and having rtol and atol being argument of this "near" function?

Reply:
you mean as an operator or as an internal function to the numpy `event` module?

Reply:
To were ever rtol is used. In other words, "rtol" should not be a property of a NumpyEvent.

Reply:
agree with this. if rtol is used in `__eq__` only I wouldn't keep it as a class attribute. if parameters can't be added to `__eq__` which I think is the case, I would maybe do `rtol = None` by default in `__eq__` and have a `.equals` that the user can pass an rtol to if they need it?